internal/manager/run.go

```diff
@@ -162,6 +162,25 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 		return err
 	}
 
+	if featureGates[gatewayFeature] {
+		setupLog.Info("Gateway API Feature Enabled, Checking Existence of CRDs")
+		crdExistsCheckers := []crdExistsChecker{
+			gatewayCRDExistsCheceker,
+			gatewayClassCRDExistsChecker,
+			httpRouteCRDExistsChecker,
+		}
+		for _, checker := range crdExistsCheckers {
+			if !checker.CRDExists(mgr.GetClient()) {
+				setupLog.Info("CRD does not exist, disable gateway feature", "resource", checker.GVR.Resource)
+				featureGates[gatewayFeature] = false
+				// REVIEW: break here?
+				break
+			}
+		}
+	}
+
+	// TODO: check alpha CRDs if GatewayAlpha enabled?
+
```

Comment:
The difference between `Gateway` and `GatewayAlpha` is that `Gateway` is on by default, so it's not the users explicit choice to enable the relevant functionality. As such, it's not as important for the `GatewayAlpha` feature to be disabled when the CRDs are missing because it requires specific action, and beyond that the failures that would ensure would be signal to the user about the issue. That said, it would probably be better if we shape that signal so I'm in favor of adding this handling, I would only suggest that the difference be we log at warning level for this one, as we know the user specifically tried to use the feature.

Reply:
The alpha resources in gateway API does not have dependency/reference of each other, and we have a `AutoHandler` to avoid errors caused by watching non-exist resources, so I agree not to disable features when CRD does not exist. Actually we do not have a warning level in our logging package, so I think a `setupLog.Info` is OK.